internal/quaiapi/api.go

```diff
@@ -321,9 +321,9 @@ func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Addre
 // * When blockNr is -1 the chain head is returned.
 // * When blockNr is -2 the pending chain head is returned.
 func (s *PublicBlockChainAPI) GetHeaderByNumber(ctx context.Context, number rpc.BlockNumber) (map[string]interface{}, error) {
-	header, err := s.b.HeaderByNumber(ctx, number)
-	if header != nil && err == nil {
-		response := RPCMarshalETHHeader(header.Header()) //TODO: mmtx this function will break once extra fields are stripped from header.
+	wo, err := s.b.BlockByNumber(ctx, number)
```

Comment:
Will this cause calls to fail when the block is not available but the header is? What is the benefit of this over the previous code?